classes/Product.php

```diff
@@ -6057,44 +6057,100 @@ public function hasCombinations()
         return !empty($attributes);
     }
 
-    public static function getIdProductAttributesByIdAttributes($id_product, $id_attributes, $find_best = false)
-    {
-        if (!is_array($id_attributes)) {
-            return 0;
-        }
-
-        $id_product_attribute =  Db::getInstance()->getValue('
-        SELECT pac.`id_product_attribute`
-        FROM `'._DB_PREFIX_.'product_attribute_combination` pac
-        INNER JOIN `'._DB_PREFIX_.'product_attribute` pa ON pa.id_product_attribute = pac.id_product_attribute
-        WHERE id_product = '.(int)$id_product.' AND id_attribute IN ('.implode(',', array_map('intval', $id_attributes)).')
-        GROUP BY id_product_attribute
-        HAVING COUNT(id_product) = '.count($id_attributes));
+    /**
+     * Get an id_product_attribute by an id_product and one or more
+     * id_attribute.
+     *
+     * e.g: id_product 8 with id_attribute 4 (size medium) and
+     * id_attribute 5 (color blue) returns id_product_attribute 9 which
+     * is the dress size medium and color blue.
+     *
+     * @param int $idProduct
+     * @param int|int[] $idAttributes
+     * @param bool $findBest
+     * @return int
+     * @throws PrestaShopException
+     */
+    public static function getIdProductAttributeByIdAttributes($idProduct, $idAttributes, $findBest = false)
+    {
+        $idProduct = (int) $idProduct;
+
+        if (!is_array($idAttributes) && is_numeric($idAttributes)) {
+            $idAttributes = array((int) $idAttributes);
+        }
+
+        if (!is_array($idAttributes) || empty($idAttributes)) {
```

Comment:
You already tests for `!is_array($idAttributes)` and build `$idAttributes` as array so you can remove this like and leave `empty` alone :)

Reply:
If we send a string for `$idAttributes`, it appears the previous test won't pass and makes this `!is_array($idAttributes)` relevant.